File: t1809e/java/assignment/assignment6/News.java
import jdk.nashorn.internal.ir.debug.ObjectSizeCalculator;

public class News implements INews {
    private int id;
    private String title;
    private String publishDate;
    private String author;
    private String content;
    private float AverageRate;

    int[] RateList = new int[3];

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getPublishDate() {
        return publishDate;
    }

    public void setPublishDate(String publishDate) {
        this.publishDate = publishDate;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public float getAverageRate() {
        return AverageRate;
    }

    public int[] getRateList() {
        return RateList;
    }

    public void setRateList(int[] rateList) {
        RateList = rateList;
    }

    @Override
    public void display() {
        System.out.println("Title: " + getTitle());
        System.out.println("PublishDate: " + getPublishDate());
        System.out.println("Author: " + getAuthor());
        System.out.println("Content: " + getContent());
        System.out.println("AverageRate: " + getAverageRate());
    }

    public void calculate(){
        int sum = 0;
        for (int i:getRateList()) {
            sum += i;
        }
        this.AverageRate = (float) sum/getRateList().length;
    }
}

File: t1809e/c#/Assignment-3-News/News.cs
using System;
using System.Linq;

namespace assignment_3_new
{
    public class News: INews
    {
        public int Id { get; set; }

        public string Title { get; set; }

        public string PublishDate { get; set; }

        public string Author { get; set; }

        public string Content { get; set; }
        
        public int[] RateList { get; set; }

        public float AverageRate { get; set; }
        
        public News()
        {
        }

        public News(int id, string title, string publishDate, string author, string content, int[] rateList)
        {
            Id = id;
            Title = title;
            PublishDate = publishDate;
            Author = author;
            Content = content;
            RateList = rateList;
        }

        public void Display()
        {
            Console.WriteLine("Title: {0, 10} | PublishDate: {1, 10} | Author: {2, 10} | Content: {3, 10} | AverageRate: {4, 10}", Title, PublishDate, Author, Content, AverageRate);
        }

        public void Calculate()
        {
            var sumRate = RateList.Sum();
            AverageRate = sumRate / RateList.Length;
        }
    }
}